### PR TITLE
chore: fix ESLint warnings

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Fixing all ESLint warnings on `main` branch

I also enabled `prefer-const` for all code blocks in `.md` files

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -6,10 +6,22 @@ module.exports = {
     "plugin:markdown/recommended",
   ],
   overrides: [
+    {
```
Comment (chaance) [inline .eslintrc.js:9]: Can we revert the changes to the config for now? We're going to review them internally as we all have some opinions with the new rules as they are.

Comment (chaance) [inline .eslintrc.js:9]: Can we revert the changes to the config for now? We're going to review them internally as we all have some opinions with the new rules as they are.

File: docs/pages/technical-explanation.md [hunk 2]
```diff
@@ -167,7 +168,7 @@ Taking our route module from before, here are a few small, but useful UX improve
 2. Focus the input when server side form validation fails
 3. Animate in the error messages
 
-```jsx nocopy lines=[4-6,8-12,21,22,28-30]
+```jsx nocopy lines=[4-6,8-12,23,25,30-34]
```
Comment (chaance) [inline docs/pages/technical-explanation.md:171]: Is this intentional? Didn't cross-check the docs site but just double checking since it's not a formatting change.

File: docs/tutorials/jokes.md [hunk 3]
```diff
@@ -2626,6 +2627,7 @@ export const action: ActionFunction = async ({
 export default function Login() {
   const actionData = useActionData<ActionData>();
   const [searchParams] = useSearchParams();
+
```
Comment (chaance) [inline docs/tutorials/jokes.md:2630]: Any reason for all the extra lines? ESLint shouldn't have an effect on formatting.

File: docs/tutorials/jokes.md [hunk 4]
```diff
@@ -3791,6 +3806,7 @@ export const action: ActionFunction = async ({
...
```
Comment (chaance) [inline docs/tutorials/jokes.md:3809]: Any reason for all the extra lines? ESLint shouldn't have an effect on formatting.

File: examples/sanity/app/root.jsx [hunk 5]
```diff
@@ -38,6 +38,7 @@ function Document({ children, title }) {
 }
 
 export default function App() {
+  // eslint-disable-next-line no-unused-vars
```
Comment (chaance) [inline examples/sanity/app/root.jsx:41]: If this is unused we should just remove it.

File: packages/remix-dev/server-build.ts [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import type { ServerBuild } from "@remix-run/server-runtime";
 
+export const assets: ServerBuild["assets"] = undefined!;
```
Comment (chaance) [inline packages/remix-dev/server-build.ts:3]: Let's revert this. Code changes are merged into `dev` until release, only docs and examples should get merged into `main`. After the release we merge everything back into `main` and `dev` from the release branch. We can live with the warning for now.

File: examples/file-and-cloudinary-upload/app/routes/cloudinary-upload.tsx [hunk 7]
```diff
@@ -42,20 +44,28 @@ export const action: ActionFunction = async ({ request }) => {
 
 export default function Index() {
   const data = useActionData<ActionData>();
+
   return (
     <>
       <Form method="post" encType="multipart/form-data">
         <label htmlFor="img-field">Image to upload</label>
+
```
Comment (chaance) [inline examples/file-and-cloudinary-upload/app/routes/cloudinary-upload.tsx:52]: Let's just remove all of the opinionated formatting changes from the PR if you don't mind, ESLint should be hands-off here 🙏

**Review by chaance — CHANGES_REQUESTED** [review]
Left some comments and requested some changes, but I think I by the end of it

File: .eslintrc.js (hunk 1 above)
Comment (MichaelDeBoey) [reply .eslintrc.js:9]: The changes to the config you did in the previous PR were either fixed (so no need for the rule anymore) or were moved to be more consistent with the rest of the config + to the code they're actually applying to.

File: examples/sanity/app/root.jsx (hunk 5 above)
Comment (MichaelDeBoey) [reply examples/sanity/app/root.jsx:41]: I thought it was used for some sanity test or so, that's why I disabled the rule here

File: docs/pages/technical-explanation.md (hunk 2 above)
Comment (MichaelDeBoey) [reply docs/pages/technical-explanation.md:171]: These weren't changed with the removal of `aria-errormessage` from `form`

**Review by LANCENET — APPROVED** [review]
Ok

**Review by LANCENET — COMMENTED** [review]
Ok

**Comment by MichaelDeBoey** [comment]
All ESLint warnings are already fixed.

`.eslintrc` changes are done in separate PRs: #2226 & #2228
